clouddriver-aws/src/main/java/com/netflix/spinnaker/clouddriver/aws/deploy/ops/CopyLastAsgAtomicOperation.java

```diff
@@ -0,0 +1,552 @@
+                        Optional.ofNullable(description.getAssociatePublicIpAddress())
+                            .orElse(associatePublicIpAddress));
+
+                    newDescription.setCooldown(
+                        Optional.ofNullable(description.getCooldown())
+                            .orElse(ancestorAsg.getDefaultCooldown()));
+                    newDescription.setEnabledMetrics(
+                        Optional.ofNullable(description.getEnabledMetrics())
+                            .orElse(
+                                ancestorAsg.getEnabledMetrics().stream()
+                                    .map(i -> i.getMetric())
+                                    .collect(Collectors.toList())));
+
+                    newDescription.setHealthCheckGracePeriod(
+                        Optional.ofNullable(description.getHealthCheckGracePeriod())
+                            .orElse(ancestorAsg.getHealthCheckGracePeriod()));
+
+                    newDescription.setHealthCheckType(
+                        Optional.ofNullable(description.getHealthCheckType())
+                            .orElse(ancestorAsg.getHealthCheckType()));
+
+                    newDescription.setSuspendedProcesses(
+                        Optional.ofNullable(description.getSuspendedProcesses())
+                            .orElse(
+                                ancestorAsg.getSuspendedProcesses().stream()
+                                    .map(i -> i.getProcessName())
+                                    .collect(Collectors.toSet())));
+
+                    newDescription.setTerminationPolicies(
+                        Optional.ofNullable(description.getTerminationPolicies())
+                            .orElse(ancestorAsg.getTerminationPolicies()));
+
+                    newDescription.setKernelId(
+                        Optional.ofNullable(description.getKernelId())
+                            .orElse(Optional.ofNullable(kernelId).orElse(null)));
+
+                    newDescription.setRamdiskId(
+                        Optional.ofNullable(description.getRamdiskId())
+                            .orElse(Optional.ofNullable(ramdiskId).orElse(null)));
+
+                    newDescription.setInstanceMonitoring(
+                        Optional.ofNullable(description.getInstanceMonitoring())
+                            .orElse(instanceMonitoring));
+                    newDescription.setEbsOptimized(
+                        Optional.ofNullable(description.getEbsOptimized()).orElse(ebsOptimized));
+
+                    newDescription.setClassicLinkVpcId(
+                        Optional.ofNullable(description.getClassicLinkVpcId())
+                            .orElse(classicLinkVPCId));
+                    newDescription.setClassicLinkVpcSecurityGroups(
+                        Optional.ofNullable(description.getClassicLinkVpcSecurityGroups())
+                            .orElse(
```

Comment:
Probably worth changing this to `orElseGet` so that the method is only invoked when needed.